src/common/file_system.cpp

```diff
@@ -244,11 +244,11 @@ string FileSystem::ExtractBaseName(const string &path) {
 	return vec[0];
 }
 
-string FileSystem::GetHomeDirectory(optional_ptr<FileOpener> opener) {
+string FileSystem::GetHomeDirectory(optional_ptr<FileOpener> opener, optional_ptr<FileOpenerInfo> info) {
 	// read the home_directory setting first, if it is set
 	if (opener) {
 		Value result;
-		if (opener->TryGetCurrentSetting("home_directory", result)) {
+		if (opener->TryGetCurrentSetting("home_directory", result, info.operator*())) {
```

Comment:
Should there be a check before referencing info here. Also can you use the operator as a prefix i.e. `*info`?